File: Kotlin/Programmers/Level 2/연속된 부분 수열의 합.kt
// 투 포인터 알고리즘
class Solution {
    fun solution(sequence: IntArray, k: Int): List<Int> {
        val answer = mutableListOf<Int>(0, sequence.lastIndex)
        
        var tmp = 0
        var right = 0
        for (i in sequence.indices) {
            while (tmp < k && right < sequence.size) {
                tmp += sequence[right]
                right += 1
            }
            
            val endIndex = right - 1
            if (tmp == k && answer[1] - answer[0] > endIndex - i) {
                answer[0] = i
                answer[1] = endIndex
            }
            
            tmp -= sequence[i]
        }
        
        return answer
    }
}

File: JavaScript/Programmers/Level 2/연속된 부분 수열의 합.js
function solution(sequence, k) {
    const answer = [0, sequence.length]
    let [tmp, right] = [0, 0]
    for (let i = 0; i < sequence.length; i++) {
        while (right < sequence.length && tmp < k) {
            tmp += sequence[right]
            right += 1
        }
        
        if (tmp === k && right - 1 - i < answer[1] - answer[0]) {
            [answer[0], answer[1]] = [i, right - 1]
        }
        
        tmp -= sequence[i]
    }
    
    return answer
}